azerothcore/azerothcore-wotlk
# feat(Core/Spells): implemented two new custom attributes to handle aura saving rule

In this PR i implemented two new custom spell attributes and fixed implicit conversation in generic SpellInfo attributes checks



## Changes Proposed:
-  Implemented initial attributes to handle aura saving (SPELL_ATTR0_CU_REJECT_AURA_SAVING, SPELL_ATTR0_CU_FORCE_AURA_SAVING)
-  Applied some code style to old code
-  Added mask stacking rules for SPELL_ATTR0_CU_NEGATIVE_EFF and SPELL_ATTR0_CU_POSITIVE_EFF
-  If Error happens on certain attributes, only attribute which errored will get skipped. This allows other flags to continue loading
-  Raise Ally control aura saving handling been moved to DB and applied SPELL_ATTR0_CU_REJECT_AURA_SAVING to this spell
## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Tested both attributes with "Retribution Aura" (rank 1 but rest of aura ranks should work same too)
- Tested both attributes with "772 Rend" (rank 1 but rest of aura ranks should work same too)

## How to Test the Changes:


Add any aura spell id with one of these attributes and check if aura gets saved or non saved (noting, SPELL_ATTR0_CU_FORCE_AURA_SAVING and SPELL_ATTR0_CU_REJECT_AURA_SAVING doesn't stack, you can pick only one)


## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
